packages/fetchai/skills/tac_control_contract/parameters.py

```diff
@@ -0,0 +1,148 @@
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains a class representing the game parameters."""
+
+import datetime
+import logging
+from typing import Any, List, Set
+
+from aea.skills.base import Model
+
+logger = logging.getLogger("aea.tac_control_skill")
+
+
+class Parameters(Model):
+    """This class contains the parameters of the game."""
+
+    def __init__(self, **kwargs):
+        """Instantiate the search class."""
+
+        self._contract_address = kwargs.pop("contract_adress", None)  # type: Any
+        self._good_ids = kwargs.pop("good_ids", [])  # type: List[int]
+
+        self._min_nb_agents = kwargs.pop("min_nb_agents", 5)  # type: int
+        self._money_endowment = kwargs.pop("money_endowment", 200)  # type: int
+        self._nb_goods = kwargs.pop("nb_goods", 5)  # type: int
+        self._tx_fee = kwargs.pop("tx_fee", 1)
+        self._base_good_endowment = kwargs.pop("base_good_endowment", 2)  # type: int
+        self._lower_bound_factor = kwargs.pop("lower_bound_factor", 1)  # type: int
+        self._upper_bound_factor = kwargs.pop("upper_bound_factor", 1)  # type: int
+        start_time = kwargs.pop("start_time", "01 01 2020  00:01")  # type: str
+        self._start_time = datetime.datetime.strptime(
+            start_time, "%d %m %Y %H:%M"
+        )  # type: datetime.datetime
+        self._registration_timeout = kwargs.pop("registration_timeout", 10)  # type: int
+        self._competition_timeout = kwargs.pop("competition_timeout", 20)  # type: int
+        self._inactivity_timeout = kwargs.pop("inactivity_timeout", 10)  # type: int
+        self._whitelist = set(kwargs.pop("whitelist", []))  # type: Set[str]
+        self._version_id = kwargs.pop("version_id", "v1")  # type: str
+        super().__init__(**kwargs)
+        now = datetime.datetime.now()
+        if now > self.registration_start_time:
+            self.context.logger.warning(
+                "[{}]: TAC registration start time {} is in the past!".format(
+                    self.context.agent_name, self.registration_start_time
+                )
+            )
+        else:
+            self.context.logger.info(
+                "[{}]: TAC registation start time: {}, and start time: {}, and end time: {}".format(
+                    self.context.agent_name,
+                    self.registration_start_time,
+                    self.start_time,
+                    self.end_time,
+                )
+            )
+
+    @property
+    def contract_address(self) -> str:
+        """The contract address of an already deployed smart-contract."""
```

Comment:
add: `assert self._contract_address is not None, "There is no deployed contract."`

Reply:
If we assert here we cannot actually check if there is a contract address from the behaviour. It doesn't pass the assertion.
`if parameters.contract_address is None:`

Reply:
So I'd create another property called:
`is_contract_deployed` which checks whether or not the address is set.

Reply:
There is no need for this since we have `contract.is_deployed` from the Contract class.